diff --git a/include/spdlog/logger-inl.h b/include/spdlog/logger-inl.h
index d01c08da..5cd01176 100644
--- a/include/spdlog/logger-inl.h
+++ b/include/spdlog/logger-inl.h
@@ -250,7 +250,7 @@ SPDLOG_INLINE void logger::err_handler_(const std::string &msg)
 #if defined(USING_R) && defined(R_R_H) // if in R environment
         REprintf("[*** LOG ERROR #%04zu ***] [%s] [%s] {%s}\n", err_counter, date_buf, name().c_str(), msg.c_str());
 #else
-        std::fprintf(stderr, "[*** LOG ERROR #%04zu ***] [%s] [%s] {%s}\n", err_counter, date_buf, name().c_str(), msg.c_str());
+        std::fprintf(stderr, "[*** LOG ERROR #%04lu ***] [%s] [%s] {%s}\n", (unsigned long)err_counter, date_buf, name().c_str(), msg.c_str());
 #endif
     }
 }
