From 95e54c03d877f89cab681e8ac3c747ca017a22ea Mon Sep 17 00:00:00 2001
From: Seth Shelnutt <Shelnutt2@gmail.com>
Date: Fri, 22 May 2020 15:29:11 -0400
Subject: [PATCH] All cloud examples should be wrapped in BUILD_TESTING

---
google/cloud/examples/CMakeLists.txt | 10 ++---
 2 files changed, 41 insertions(+), 39 deletions(-)

diff --git a/google/cloud/examples/CMakeLists.txt b/google/cloud/examples/CMakeLists.txt
index 5af4627b4..dc052168b 100644
--- a/google/cloud/examples/CMakeLists.txt
+++ b/google/cloud/examples/CMakeLists.txt
@@ -15,13 +15,13 @@
 # ~~~

 # Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
+if (BUILD_TESTING)
+    include(SelectMSVCRuntime)

-add_executable(gcs2cbt gcs2cbt.cc)
-target_link_libraries(gcs2cbt bigtable_client storage_client
+    add_executable(gcs2cbt gcs2cbt.cc)
+    target_link_libraries(gcs2cbt bigtable_client storage_client
                       google_cloud_cpp_grpc_utils)
-google_cloud_cpp_add_common_options(gcs2cbt)
-if (BUILD_TESTING)
+    google_cloud_cpp_add_common_options(gcs2cbt)
     add_test(NAME gcs2cbt COMMAND gcs2cbt)
     set_tests_properties(
         gcs2cbt PROPERTIES LABELS
-- 
2.26.2

