From 4749df2b03e70218870dab48b83d090af2451027 Mon Sep 17 00:00:00 2001
From: Seth Shelnutt <Shelnutt2@gmail.com>
Date: Fri, 22 May 2020 14:06:41 -0400
Subject: [PATCH] Add option to disable tests

---
 CMakeLists.txt       | 1 +
 super/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9275541b..0d0058b27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,6 +143,7 @@ mark_as_advanced(GOOGLE_CLOUD_CPP_ENABLE_GRPC_UTILS)
 
 # Enable testing in this directory so we can do a top-level `make test`. This
 # also includes the BUILD_TESTING option, which is on by default.
+option(BUILD_TESTING "" OFF)
 include(CTest)
 
 # Each subproject adds dependencies to this target to have their docs generated.
2.26.2
