diff --git a/c++/src/kj/windows-sanity.h b/c++/src/kj/windows-sanity.h
index 894ec1d7..b2c93678 100644
--- a/c++/src/kj/windows-sanity.h
+++ b/c++/src/kj/windows-sanity.h
@@ -48,17 +48,24 @@
 // now, we use `#pragma once` to tell the compiler never to include this file again.
 #pragma once

-namespace win32 {
+namespace kj_win32_workarounds {
+  // Namespace containing constant definitions intended to replace constants that are defined as
+  // macros in the Windows headers. Do not refer to this namespace directly, we'll import it into
+  // the global scope below.
+
+#ifdef ERROR  // This could be absent if e.g. NOGDI was used.
   const auto ERROR_ = ERROR;
 #undef ERROR
   const auto ERROR = ERROR_;
+#endif

   typedef VOID VOID_;
 #undef VOID
   typedef VOID_ VOID;
 }

-using win32::ERROR;
-using win32::VOID;
+// Pull our constant definitions into the global namespace -- but only if they don't already exist
+// in the global namespace.
+using namespace kj_win32_workarounds;

 #endif
