diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f4ce16a03..ef5cb2d87a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.10)
 project("Cap'n Proto Root" CXX)
 include(CTest)
 
diff --git a/c++/CMakeLists.txt b/c++/CMakeLists.txt
index b465052745..2ef15397df 100644
--- a/c++/CMakeLists.txt
+++ b/c++/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.10)
 project("Cap'n Proto" CXX)
 set(VERSION 1.1.0)
 
diff --git a/c++/samples/CMakeLists.txt b/c++/samples/CMakeLists.txt
index 039b1df5e7..2a5349e9d9 100644
--- a/c++/samples/CMakeLists.txt
+++ b/c++/samples/CMakeLists.txt
@@ -16,7 +16,7 @@
 #      cmake ../c++/samples
 #      cmake --build .
 
 project("Cap'n Proto Samples" CXX)
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.10)
 
 find_package(CapnProto CONFIG REQUIRED)
diff --git a/c++/src/kj/CMakeLists.txt b/c++/src/kj/CMakeLists.txt
index 980c53e34c..cacadec58b 100644
--- a/c++/src/kj/CMakeLists.txt
+++ b/c++/src/kj/CMakeLists.txt
@@ -78,8 +78,7 @@ set(kj-std_headers
 )
 add_library(kj ${kj_sources})
 add_library(CapnProto::kj ALIAS kj)
-# TODO(cleanup): Use cxx_std_14 once it's safe to require cmake 3.8.
-target_compile_features(kj PUBLIC cxx_generic_lambdas)
+target_compile_features(kj PUBLIC cxx_std_14)
 
 if(UNIX AND NOT ANDROID)
   target_link_libraries(kj PUBLIC pthread)
