diff --git a/CMakeLists.txt b/CMakeLists.txt
index dfb441e..71ad51b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,6 +157,16 @@ elseif(WIN32)
   target_compile_definitions(azure-storage-lite PRIVATE azure_storage_lite_EXPORTS NOMINMAX)
 endif()

+#openssl3 md5 deprecation mitigation
+if(UNIX)
+  #sledge hammer approach
+  target_compile_options(azure-storage-lite PRIVATE "-Wno-deprecated-declarations")
+endif()
+if(MSVC)
+  #sledge hammer approach
+  target_compile_options(azure-storage-lite PRIVATE "/wd4996")
+endif()
+
 if(BUILD_TESTS)
   find_package(Catch2)
   if(NOT Catch2_FOUND)
