From 4e58dda184d9dbe405a8a21ad6aa9a1e46057831 Mon Sep 17 00:00:00 2001
From: Joe maley <joe@tiledb.io>
Date: Tue, 18 Feb 2020 15:54:40 -0500
Subject: [PATCH 1/2] remove uuid dependency

Signed-off-by: Isaiah Norton <ihnorton@users.noreply.github.com>
---
 CMakeLists.txt  | 10 +++++-----
 src/utility.cpp |  9 ++++++++-
 2 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dfb441e..cefd897 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,11 +147,11 @@ if (UNIX)
     list(APPEND EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
   endif()

-  if(NOT APPLE)
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
-    list(APPEND EXTRA_LIBRARIES PkgConfig::uuid)
-  endif()
+#  if(NOT APPLE)
+#    find_package(PkgConfig REQUIRED)
+#    pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
+#    list(APPEND EXTRA_LIBRARIES PkgConfig::uuid)
+#  endif()
 elseif(WIN32)
   list(APPEND EXTRA_LIBRARIES rpcrt4 bcrypt)
   target_compile_definitions(azure-storage-lite PRIVATE azure_storage_lite_EXPORTS NOMINMAX)
diff --git a/src/utility.cpp b/src/utility.cpp
index 747fcb1..07593e0 100644
--- a/src/utility.cpp
+++ b/src/utility.cpp
@@ -8,7 +8,7 @@
 #define WIN32_LEAN_AND_MEAN
 #include <Windows.h>
 #else
-#include <uuid/uuid.h>
+//#include <uuid/uuid.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <fcntl.h>
@@ -26,6 +26,7 @@ namespace azure {  namespace storage_lite {
         return str;
     }

+/*
     std::string get_uuid()
     {
         std::string res;
@@ -46,6 +47,12 @@ namespace azure {  namespace storage_lite {

         return res;
     }
+*/
+
+    std::string get_uuid() {
+      std::cerr << "azure::storage_lite::get_uuid() unimplemented" << std::endl;
+      exit(1);
+    }

     bool create_or_resize_file(const std::string& path, unsigned long long length) noexcept
     {
--
2.22.0.windows.1


From e9908d8c1956a05f5b33d8ffecd971c2b437e5aa Mon Sep 17 00:00:00 2001
From: Jinming Hu <jinmhu@microsoft.com>
Date: Sun, 21 Jun 2020 22:44:31 +0800
Subject: [PATCH 2/2] Fix build error on VS2019

---
 src/base64.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/base64.cpp b/src/base64.cpp
index e6dead7..0b3e18e 100644
--- a/src/base64.cpp
+++ b/src/base64.cpp
@@ -1,5 +1,6 @@
 #include <array>
 #include <cstring>
+#include <stdexcept>

 #include "base64.h"

--
2.22.0.windows.1

