diff --git a/src/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp b/src/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp
index b8ab519..7cd91e9 100644
--- a/src/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp
+++ b/src/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp
@@ -19,6 +19,9 @@
 
 #include <Windows.h>
 #include <winhttp.h>
+#ifdef __MINGW32__
+#include <WinSock2.h>
+#endif
 #include <mstcpip.h> // for tcp_keepalive
 #include <sstream>
 #include <iostream>
