diff --git a/google/cloud/google_cloud_cpp_common.cmake b/google/cloud/google_cloud_cpp_common.cmake
index 1a14c77..4e8058e 100644
--- a/google/cloud/google_cloud_cpp_common.cmake
+++ b/google/cloud/google_cloud_cpp_common.cmake
@@ -271,9 +271,6 @@ set(google_cloud_cpp_mocks_hdrs
     mocks/mock_stream_range.h)
 export_list_to_bazel("google_cloud_cpp_mocks.bzl" "google_cloud_cpp_mocks_hdrs"
                      YEAR "2022")
-target_link_libraries(
-    google_cloud_cpp_mocks INTERFACE google-cloud-cpp::common GTest::gmock_main
-                                     GTest::gmock GTest::gtest)
 set_target_properties(google_cloud_cpp_mocks PROPERTIES EXPORT_NAME
                                                         google-cloud-cpp::mocks)
 target_include_directories(
diff --git a/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake b/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake
index a12bb20..eb3182f 100644
--- a/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake
+++ b/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake
@@ -219,9 +219,6 @@ set(google_cloud_cpp_storage_grpc_mocks_hdrs
     mocks/mock_async_writer_connection.h)
 export_list_to_bazel("google_cloud_cpp_storage_grpc_mocks.bzl"
                      "google_cloud_cpp_storage_grpc_mocks_hdrs" YEAR "2023")
-target_link_libraries(
-    google_cloud_cpp_storage_grpc_mocks
-    INTERFACE google-cloud-cpp::experimental-storage_grpc GTest::gmock)
 set_target_properties(
     google_cloud_cpp_storage_grpc_mocks
     PROPERTIES EXPORT_NAME "google-cloud-cpp::experimental-storage_grpc_mocks")
