diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3b293781..9fb77d3a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,7 +165,7 @@ if(BUILD_TESTS)
   endif()
 endif()
 
-target_include_directories(azure-storage-lite PUBLIC ${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIRS} PRIVATE ${EXTRA_INCLUDE_DIRS})
+target_include_directories(azure-storage-lite PUBLIC ${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIR} PRIVATE ${EXTRA_INCLUDE_DIRS})
 target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EXTRA_LIBRARIES})
 
 if(MSVC)
