From f6633edbbd696e82c052a794f902671538b773c8 Mon Sep 17 00:00:00 2001
From: Joe maley <joe@tiledb.io>
Date: Tue, 18 Feb 2020 15:54:40 -0500
Subject: [PATCH] remove uuid dependency

---
 CMakeLists.txt  | 10 +++++-----
 src/utility.cpp |  9 ++++++++-
 2 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f32bc44..3c94347 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,11 +146,11 @@ if (UNIX)
     list(APPEND EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
   endif()
 
-  if(NOT APPLE)
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
-    list(APPEND EXTRA_LIBRARIES PkgConfig::uuid)
-  endif()
+#  if(NOT APPLE)
+#    find_package(PkgConfig REQUIRED)
+#    pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
+#    list(APPEND EXTRA_LIBRARIES PkgConfig::uuid)
+#  endif()
 elseif(WIN32)
   list(APPEND EXTRA_LIBRARIES rpcrt4 bcrypt)
 endif()
diff --git a/src/utility.cpp b/src/utility.cpp
index 95dcdfc..999669a 100644
--- a/src/utility.cpp
+++ b/src/utility.cpp
@@ -8,7 +8,7 @@
 #define WIN32_LEAN_AND_MEAN
 #include <Windows.h>
 #else
-#include <uuid/uuid.h>
+//#include <uuid/uuid.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <fcntl.h>
@@ -26,6 +26,7 @@ namespace azure {  namespace storage_lite {
         return str;
     }
 
+/*
     std::string get_uuid()
     {
         std::string res;
@@ -46,6 +47,12 @@ namespace azure {  namespace storage_lite {
 
         return res;
     }
+*/
+
+    std::string get_uuid() {
+      std::cerr << "azure::storage_lite::get_uuid() unimplemented" << std::endl;
+      exit(1);
+    }
 
     bool create_or_resize_file(const std::string& path, unsigned long long length) noexcept
     {
-- 
2.17.1

